src/Avalonia.Base/Data/Core/IBinding2.cs

```diff
@@ -0,0 +1,9 @@
+﻿namespace Avalonia.Data.Core;
+
+internal interface IBinding2 : IBinding
+{
+    BindingExpressionBase Instance(
```

Comment:
@grokys do I understood it correctly, that InstancedBinding is now obsolete with new bindings, as BindingExpression works as a specific instance of a binding?

And I wonder what's a plan with dealing IBinding2.
In general, it looks like you can combine it into a single interface (since IBinding is not client implementable). But keep old `Instance` method with Obsolete attribute.

Reply:
Yeah `IBinding2` is a total hack, ideally it'd all be implemented as internal code in a `BindingBase` class but too late for that now. I need to add some comments on how it should be refactored for 12.0.

Reply:
Just to be clear, this type of thing will be cleaned up for 12.0 as a breaking change, right?

Reply:
we should add later on comments `// TODO12: Remove me` otherwise we may miss it.